### PR TITLE
[Term Entry] Go String Functions Trim()

### PR DESCRIPTION
**Pull request by 4hbab**
### Description

- Added strings.Trim() term for go
- Closes Issue #2633 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/strings/terms/trim/trim.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+Title: 'Trim()'
+Description: 'Removes leading and trailing specified characters from a string.'
```
Comment (caupolicandiaz) [inline content/go/concepts/strings/terms/trim/trim.md:3]:
```suggestion
Description: 'Removes leading and trailing characters from a string based on the given input.'
```

File: content/go/concepts/strings/terms/trim/trim.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Functions'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Trim()`** function removes leading and trailing specified characters from a string. The `strings` library must be imported in order to use this function.
```
Comment (caupolicandiaz) [inline content/go/concepts/strings/terms/trim/trim.md:15]:
```suggestion
The **`Trim()`** function removes leading and trailing characters from a string based on the given input. The `strings` library must be imported in order to use this function.
```

File: content/go/concepts/strings/terms/trim/trim.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Syntax
+
+```pseudo
+trimmed := strings.Trim(str, cutset)
+```
+
+Where `str` is the input string from which the leading and trailing characters are to be removed, `cutset` is a string containing the characters to be removed, and `trimmed` is the resulting string after trimming.
```
Comment (caupolicandiaz) [inline content/go/concepts/strings/terms/trim/trim.md:23]: I would suggest reformatting this as a list-
- `str`: The input string from which the leading and trailing characters are to be removed.
- etc

File: content/go/concepts/strings/terms/trim/trim.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+The output will be:
+
+```shell
+Hello, World!
+```
+
+## Codebyte Example
+
+The following runnable example shows the usage of the `Trim()` function to remove specific characters from a string:
```
Comment (caupolicandiaz) [inline content/go/concepts/strings/terms/trim/trim.md:52]:
```suggestion
The following runnable example uses the `Trim()` function to remove specific characters from a string:
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@4hbab I've added a few comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by 4hbab** [comment]
@caupolicandiaz I've made the suggested changes

**Review by caupolicandiaz — APPROVED** [review]
@4hbab Looks good. I'm passing this on to a second review.

**Review by SSwiniarski — APPROVED** [review]
@4hbab Looks good to merge.

**Comment by SSwiniarski** [comment]
@4hbab Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/go/strings/trim

image

**Pull request merged**